### DISCUSSION TITLE
Program Selection (Dishwasher)

### DISCUSSION THREAD

**Discussion by outsharded** [thoukydides/homebridge-homeconnect#296] [Ideas]
Currently, the program selection is made of switches, which seems confusing. Have you considered using a selection in HomeKit?
![menu](https://github.com/user-attachments/assets/064cf64e-fb70-4e09-9768-e0cfa944aada)

This could make program selection much easier.

**Comment by thoukydides** [comment]
Yes, I considered implementing the program selection as a `Television` service, and experimented with it briefly. However, I rejected it because (i) it confuses Siri since it thinks it is controlling a TV (and I have a Harmony Hub actually controlling a TV), and (ii) the programs are not mutually exclusive and that is not supported by the `Television` service.

This has been discussed previously, e.g. in #58.
